task.go

```diff
@@ -0,0 +1,71 @@
+package mesg
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/mesg-foundation/core/api/service"
+)
+
+var errOnlyOneOutputKey = errors.New("there should be only one output key in response")
+
+// Task represents a MESG task.
+type Task struct {
+	name    string
+	handler func(*Request) Response
+}
+
+// NewTask creates a task with name, handler executed when a matching task request is received.
+func NewTask(name string, handler func(*Request) Response) Task {
+	t := Task{
+		name:    name,
+		handler: handler,
+	}
+	return t
+}
+
+// Key is the output key of task.
+type Key string
+
+// Data is an event data or output data of a task.
+type Data interface{}
+
+// Response is a task response.
+type Response map[Key]Data
```

Comment:
`Response` should be rename by `Output` and its data's structure could be improve.
What about using OutputKey and OutputData?
A task execution can only return one output at a time even if the task's definition declare many output.

Reply:
Could be:
```go
type Output struct {
  Key string
  Data interface{}
}
```

Reply:
yes, I agree according to the service API description. but we should totally drop Response type because it is a map. Task executions should return with `(string, interface{})`, where _string_ is the output key and _interface{}_ is output data. _interface{}_ lets us to return structs or maps as task outputs.